src/utils/async.js

```diff
@@ -2,7 +2,7 @@
 import { nullFunction } from '../nullObjects';
 
 /** Like setTimeout(..., 0), but returns a Promise of the result. */
-export function delay<T>(callback: () => T): Promise<T> {
+export function delay<T>(callback: () => void | Promise<any>): Promise<T> {
```

Comment:
I am not excited about adding an `any` here. In general, I want to avoid adding `any`.

What's the specific issue you ran into that made you want to make this type looser? Let's debug and find a better solution. I'm sure there are ways I'd be happy to adjust this type -- but using `any` is not going to be one of them. :stuck_out_tongue: